### PR TITLE
Check append return code

### LINKED ISSUE #1761: What's the behavior of SendRpcResponse in memory constrained env?
It seems `SendRpcResponse` contains heap allocations (`butil::IOBuf::append`) which returns `-1` if malloc fails. This error code is silently dropped when building response. Do we have defined behavior when response is partially built and sent back to client?

ref:

https://github.com/apache/incubator-brpc/blob/master/src/brpc/policy/baidu_rpc_protocol.cpp#L222-L229

btw, can we mark `SendRpcResponse` as **noexcept** ?

### PR DESCRIPTION
**Pull request by wwbmmm**
Fix #1761

### CODE REVIEW

File: src/brpc/policy/baidu_rpc_protocol.cpp [hunk 1]
```diff
@@ -80,11 +80,11 @@ static void SerializeRpcHeaderAndMeta(
         ::google::protobuf::io::CodedOutputStream coded_out(&arr_out);
         meta.SerializeWithCachedSizes(&coded_out); // not calling ByteSize again
         CHECK(!coded_out.HadError());
-        out->append(header_and_meta, sizeof(header_and_meta));
+        CHECK_EQ(0, out->append(header_and_meta, sizeof(header_and_meta)));
```
Comment (amosbird) [inline src/brpc/policy/baidu_rpc_protocol.cpp:83]: IIUC this is just a assertion and will only be effective in debug build. Perhaps we should change the return type of `SerializeRpcHeaderAndMeta` into `bool` and handle it properly, similar to what we did for `SerializeAsCompressedData`.

Comment (wwbmmm) [reply src/brpc/policy/baidu_rpc_protocol.cpp:83]: // The CHECK(condition) macro is active in both debug and release builds and
// effectively performs a LOG(FATAL) which terminates the process and
// generates a crashdump unless a debugger is attached.

https://github.com/apache/incubator-brpc/blob/master/src/butil/logging.h#L122

Comment (amosbird) [reply src/brpc/policy/baidu_rpc_protocol.cpp:83]: Ok... It's interesting to see a release assertion macro) However, this error is recoverable. Resorting to `std::terminate` seems to be an overkill.

Comment (serverglen) [reply src/brpc/policy/baidu_rpc_protocol.cpp:83]: LGTM

**Pull request merged**